dbms/src/Columns/ColumnString.cpp

```diff
@@ -484,29 +495,47 @@ void ColumnString::getPermutationWithCollationImpl(
 
 void ColumnString::countSerializeByteSizeForCmp(
     PaddedPODArray<size_t> & byte_size,
+    const NullMap * nullmap,
     const TiDB::TiDBCollatorPtr & collator) const
 {
-    if likely (collator != nullptr)
+    // Skip decoding collator for bin collaotr so we can avoid counting code points, which may be slow.
+    if (collator != nullptr && collator->sortKeyReservedSpaceMultipler() > 1)
```

Comment:
When `sortKeyReservedSpaceMultipler` returns number that is not greater than 1?

Reply:
for bin collator, sortKeyReservedSpaceMultipler is 1.
for others(general ci, UCACI...), sortKeyReservedSpaceMultipler is greater than 1.